engine/table/src/main/java/io/deephaven/engine/util/input/InputTableValidationException.java

```diff
@@ -0,0 +1,202 @@
+ */
+public class InputTableValidationException extends UncheckedDeephavenException {
+    /**
+     * An error indicating that invalid data was entered; with an optional row position and column name.
+     *
+     * <p>
+     * Implementers of {@link InputTableUpdater} may use {@link StructuredErrorImpl} as an implementation.
+     * </p>
+     * .
+     */
+    public interface StructuredError {
+        /**
+         * @return the row position in the table to add that was invalid; or {@link OptionalLong#empty() empty} if
+         *         unknown.
+         */
+        OptionalLong getRow();
+
+        /*
+         * @return the column name in the table to add that was invalid; or {@link Optional#empty() empty} if unknown.
+         */
+        Optional<String> getColumn();
+
+        /**
+         * @return the error message for this validation failure.
+         */
+        String getMessage();
+    }
+
+    private final List<StructuredError> errors;
+
+    /**
+     * Create an InputTableValidationException with the given message.
+     *
+     * <p>
+     * No structured errors are supplied.
+     * </p>
+     *
+     * @param message the error message
+     */
+    public InputTableValidationException(final String message) {
+        super(message);
+        errors = Collections.emptyList();
+    }
+
+    /**
+     * Create an InputTableValidationException with the given message.
+     *
+     * @param message the error message
+     * @param errors the structured errors for reporting to callers
+     */
+    public InputTableValidationException(final String message, final List<StructuredError> errors) {
+        super(message);
+        this.errors = errors;
+    }
+
+    /**
+     * Create an InputTableValidationException with the given errors.
+     *
+     * <p>
+     * The exception message is generated from the list of errors, attempting to provide a user-actionable summary for
+     * callers that do not interpret structured errors.
+     * </p>
+     *
+     * @param errors the structured errors for reporting to callers
+     */
+    public InputTableValidationException(final List<StructuredError> errors) {
+        this(generateMessage(errors), errors);
+    }
+
+    private static String generateMessage(final List<StructuredError> errors) {
+        if (errors.isEmpty()) {
+            return "Unknown InputTable validation error.";
+        }
+        final StringBuilder message = new StringBuilder();
+        if (errors.size() == 1) {
+            final StructuredError firstError = errors.get(0);
+            message.append("Input Table validation error");
+            return formatOneMessage(message, firstError).toString();
+        }
+        final Map<String, List<StructuredError>> errorMap = errors.stream()
+                .collect(Collectors.toMap(StructuredError::getMessage, e -> new ArrayList<>(List.of(e)), (l1, l2) -> {
+                    l1.addAll(l2);
+                    return l1;
+                }));
```

Comment:
I think this is just

```suggestion
                .collect(Collectors.groupingBy(StructuredError::getMessage));
```